pkg/controller/configmap/configmap_controller.go

```diff
@@ -0,0 +1,175 @@
+			logger.Info("ConfigMap not found, ignoring since object must be deleted")
+			return ctrl.Result{}, nil
+		}
+		logger.Error(err, "Failed to get ConfigMap")
+		return ctrl.Result{}, err
+	}
+
+	logger.Info("Processing ConfigMap for MAC range update")
+
+	// Extract and validate new MAC ranges from ConfigMap
+	rangeConfig, err := r.extractRangesFromConfigMap(configMap)
+	if err != nil {
+		logger.Error(err, "Failed to extract ranges from ConfigMap",
+			"data", configMap.Data)
+		return ctrl.Result{}, fmt.Errorf("invalid ConfigMap data: %v", err)
+	}
+
+	logger.V(1).Info("Extracted new MAC ranges from ConfigMap",
+		"newRangeStart", rangeConfig.Start.String(),
+		"newRangeEnd", rangeConfig.End.String())
+
+	// Check if the new ranges are different from current ranges
+	currentStart, currentEnd := r.PoolManager.GetCurrentRanges()
+	if rangeConfig.Start.String() == currentStart.String() && rangeConfig.End.String() == currentEnd.String() {
+		logger.Info("MAC address ranges unchanged, skipping update",
+			"currentRangeStart", currentStart.String(),
+			"currentRangeEnd", currentEnd.String())
+		return ctrl.Result{}, nil
+	}
+
+	logger.Info("MAC address ranges changed, applying update",
+		"oldRangeStart", currentStart.String(),
+		"oldRangeEnd", currentEnd.String(),
+		"newRangeStart", rangeConfig.Start.String(),
+		"newRangeEnd", rangeConfig.End.String())
+
+	// Apply the new ranges to PoolManager
+	if err := r.PoolManager.UpdateRanges(rangeConfig.Start, rangeConfig.End); err != nil {
+		logger.Error(err, "Failed to update MAC address ranges in PoolManager",
+			"rangeStart", rangeConfig.Start.String(),
+			"rangeEnd", rangeConfig.End.String())
+		return ctrl.Result{}, fmt.Errorf("failed to apply new ranges: %v", err)
+	}
+
+	logger.V(1).Info("Successfully updated MAC address ranges",
+		"rangeStart", rangeConfig.Start.String(),
+		"rangeEnd", rangeConfig.End.String())
+
+	return ctrl.Result{}, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *ConfigMapReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	// Create predicate to only watch our specific ConfigMap
+	configMapPredicate := predicate.NewPredicateFuncs(func(object client.Object) bool {
+		return object.GetName() == names.MAC_RANGE_CONFIGMAP && object.GetNamespace() == r.ConfigMapNamespace
+	})
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named("mac-range-configmap").
+		For(&corev1.ConfigMap{}).
+		WithEventFilter(configMapPredicate).
+		Complete(r)
+}
+
+// extractRangesFromConfigMap parses MAC ranges from ConfigMap data
+func (r *ConfigMapReconciler) extractRangesFromConfigMap(cm *corev1.ConfigMap) (*RangeConfig, error) {
+	// Validate ConfigMap has the expected keys
+	startStr, ok := cm.Data["RANGE_START"]
+	if !ok {
+		return nil, fmt.Errorf("RANGE_START not found in ConfigMap")
+	}
+
+	endStr, ok := cm.Data["RANGE_END"]
+	if !ok {
+		return nil, fmt.Errorf("RANGE_END not found in ConfigMap")
+	}
+
+	// Parse and validate MAC addresses
```

Comment:
Is it worth it to add constants for RANGE_START and RANGE_END ? I see them again somewhere in the code

Reply:
Let's do it on a follow up PR